tests/integrationv2/providers.py

```diff
@@ -82,37 +121,35 @@ class S2N(Provider):
     """
     The S2N provider translates flags into s2nc/s2nd command line arguments.
     """
-
-    # Describe all ciphers that fall into the "default_tls13" preference list.
-    default_tls13 = [
-        Ciphersuites.TLS_CHACHA20_POLY1305_SHA256,
-        Ciphersuites.TLS_AES_128_GCM_256,
-        Ciphersuites.TLS_AES_256_GCM_384
-    ]
-
     def __init__(self, options: ProviderOptions):
-        self.ready_to_send_marker = None
+        self.ready_to_send_input_marker = None
         Provider.__init__(self, options)
 
     def setup_client(self, options: ProviderOptions):
         """
         Using the passed ProviderOptions, create a command line.
         """
         cmd_line = ['s2nc', '-e']
-        if options.cipher is not None:
-            if options.cipher in S2N.default_tls13:
-                cmd_line.extend(['-c', 'default_tls13'])
-            else:
-                cmd_line.extend(['-c', 'default'])
+
+        # This is the last thing printed by s2nc before it is ready to send/receive data
+        self.ready_to_send_input_marker = 'Cipher negotiated:'
+
         if options.use_session_ticket is False:
             cmd_line.append('-T')
+
         if options.insecure is True:
             cmd_line.append('--insecure')
-        if options.tls13 is True:
+        else:
+            if options.cert is not None:
+                cmd_line.extend(['-f', options.cert])
+
+        if options.protocol == Protocols.TLS13:
             cmd_line.append('--tls13')
-        cmd_line.extend([options.host, options.port])
+            cmd_line.extend(['-c', 'test_all_tls13'])
+        else:
+            cmd_line.extend(['-c', 'test_all'])
```

Comment:
`test_all` contains all ciphers present in s2n including the ciphers present in `test_all_tls13`. 
Do we want to be more specific and use `test_all_tls12` when we want to test TLS1.2 version?

Reply:
If we had more fine grained control I would be ok with specifying a preference set based on the protocol. But we may be testing ciphers with protocols below TLS1.2, so I don't want to leave those out.